test/GeneratorTests.cpp

```diff
@@ -0,0 +1,25 @@
+#include <gtest/gtest.h>
+#include "Generator.hpp"
+#include <set>
+#include <cmath>
+
+struct GeneratorTests : public ::testing::Test
+{
+};
+
+TEST_F(GeneratorTests, checkAllPermutationOfSingleVector)
```

Comment:
It's rather checking the number of generated permutations, so this test name should describe it.

Reply:
Ok